pkg/controller/machineautoscaler/machinetarget_test.go

```diff
@@ -418,6 +418,18 @@ func TestUpdatingScaleFromZeroAnnotations(t *testing.T) {
 				annotationsutil.GpuTypeKey:            "nvidia.com/gpu",
 			},
 		},
+		{
+			name: "Supplying old GPU annotations and adding upstream annotations with empty string if count is 0",
+			expectedNewAnnotation: map[string]string{
+				annotationsutil.GpuCountKey:           "0",
+				annotationsutil.GpuCountKeyDeprecated: "0",
+				annotationsutil.GpuTypeKey:            "",
+			},
+			suppliedAnnotations: map[string]string{
+				annotationsutil.GpuCountKeyDeprecated: "0",
+				annotationsutil.GpuTypeKey:            "",
```

Comment:
i think we should remove this from the supplied since it wouldn't be there.

Reply:
Thanks Mike, good catch. I removed it from the case above for the same reason